cloudinit/sources/DataSourceVMware.py

```diff
@@ -53,6 +56,63 @@
 WAIT_ON_NETWORK_IPV4 = "ipv4"
 WAIT_ON_NETWORK_IPV6 = "ipv6"
 
+SUPPORTED_UPDATE_EVENTS_GUEST_INFO_KEY = "cloudinit.update-events.supported"
+ENABLED_UPDATE_EVENTS_GUEST_INFO_KEY = "cloudinit.update-events.enabled"
+
+# Support network reconfiguration for the following events:
+# TODO(akutz) Add support for METADATA_CHANGE and USER_REQUEST when
+#             those events are supported by Cloud-Init.
+SUPPORTED_UPDATE_EVENTS = {
+    EventScope.NETWORK: {
+        EventType.BOOT,
+        EventType.BOOT_NEW_INSTANCE,
+        EventType.HOTPLUG,
+    }
+}
+
+# By default this datasource only configures the network when
+# booting a new instance or when a NIC is added or removed. If
+# a control plane like VM Service wishes the network to be
+# reconfigured on each boot, it should send in vendordata that
+# overrides the default update events.
+DEFAULT_UPDATE_EVENTS = {
+    EventScope.NETWORK: {
+        EventType.BOOT_NEW_INSTANCE,
+        EventType.HOTPLUG,
+    }
+}
+
+# Only trigger hook-hotplug on NICs with VMware drivers. Avoid
+# triggering the hook on Docker virtual NICs, etc.
+#
+# Please note, there is no way for this datasource to work
+# out-of-the-box with SR-IOV NICs since their driver IDs are not known
+# ahead of time. An SR-IOV NIC's driver is specific to the physical
+# NIC. Users that want to support hotplug events for SR-IOV NICs will
+# need to set "metadata.network-drivers" to a list of strings that
+# includes the name of the driver used by the SR-IOV NIC.
+NETWORK_DRIVERS = "network-drivers"
```

Comment:
Is this string used anywhere?

Reply:
It was meant to be. I just updated the constants to use it.